astropy/io/fits/convenience.py

```diff
@@ -489,7 +514,9 @@ def table_to_hdu(table):
 
             col.null = fill_value.astype(table[col.name].dtype)
     else:
-        table_hdu = BinTableHDU.from_columns(np.array(table.filled()))
+        table_hdu = BinTableHDU.from_columns(np.array(table.filled()), header=hdr)
+
+    coord_meta = table.meta.pop('__coordinate_columns__', None)
```

Comment:
After further thought, I think I agree with having it on the table, and it seems fine to give a different name, since it doesn't really have the same purpose as `__mixin_columns__` (and doesn't contain the same information). And in any case, it is an implementation detail that we can change later.